src/api/customAPI.ts

```diff
@@ -32,9 +28,30 @@ const axiosApi = ({ options }: any) => {
       return Promise.reject(error);
     }
   );
+};
 
-  instance.defaults.timeout = 2000; // 2초
+const createInstance = (headers?: any) => {
+  const instance = axios.create({
+    baseURL: API_BASE_URL,
+    timeout: 2000,
+    headers,
+  });
```

Comment:
토큰을 파라미터로 받는 대신, axios 인스턴스를 생성하는 메소드가 토큰을 확인하고 그에맞는 인스턴스를 리턴하는 방식은 어떨까용?

```suggestion
  const instance = TOKEN ? axios.create({
    baseURL: API_BASE_URL,
    timeout: 2000,
    header:{
       Authorization: `Bearer ${TOKEN}`
    },
  }); : axios.create({
    baseURL: API_BASE_URL,
    timeout: 2000,
  });
```

Reply:
좋은 생각입니다
```ts
const createInstance = (isFormData?: boolean) => {
  const instance = token
    ? axios.create({
        baseURL: API_BASE_URL,
        timeout: 2000,
        headers: {
          Authorization: `Bearer ${token}`,
          'Content-Type': isFormData ? 'multipart/form-data' : 'application/json',
        },
      })
    : axios.create({
        baseURL: API_BASE_URL,
        timeout: 2000,
      });
  setInterceptors(instance);
  return instance;
};

const axiosApi = (isFormData?: boolean) => {
  return createInstance(isFormData);
};
```

의견 반영해서 코드를 변경해봤는데 api.ts를 봤을 때 헤더에 어떤게 담겨 갔을지 예측이 안될 거 같은 우려와

isFormData가 들어간 경우 true를 인자를 넘겨줘야 하는데 일관성이 떨어져서 가독성을 해칠 거 같아요 ..!

그래서 우선 PR보낸 부분을 사용해보고 customAPI에서 추가해야할 부분이 생기면 그 때 리팩토링 고려해볼게요



### 바뀌게 된다면 변경 될 전체 코드
```ts
/* eslint-disable no-console */
/* eslint-disable @typescript-eslint/no-explicit-any */
import axios, { AxiosInstance } from 'axios';

const API_BASE_URL = process.env.NEXT_PUBLIC_API_BASE_URL;

const token = 'my token'; // 토큰은 저장하면 로직을 바꾸도록 할게요

const setInterceptors = (instance: AxiosInstance) => {
  instance.interceptors.response.use(
    (response) => {
      console.log('interceptor > response', response);
      return response;
    },
    (error) => {
      console.log('interceptor > error', error);
      return Promise.reject(error);
    }
  );

  instance.interceptors.request.use(
    (request) => {
      console.log('interceptor > request', request);
      return request;
    },
    (error) => {
      console.log('interceptor > error', error);
      return Promise.reject(error);
    }
  );
};

const createInstance = (isFormData?: boolean) => {
  const instance = token
    ? axios.create({
        baseURL: API_BASE_URL,
        timeout: 2000,
        headers: {
          Authorization: `Bearer ${token}`,
          'Content-Type': isFormData ? 'multipart/form-data' : 'application/json',
        },
      })
    : axios.create({
        baseURL: API_BASE_URL,
        timeout: 2000,
      });
  setInterceptors(instance);
  return instance;
};

const axiosApi = (isFormData?: boolean) => {
  return createInstance(isFormData);
};

export const instance = axiosApi();

```

Reply:
> isFormData가 들어간 경우 true를 인자를 넘겨줘야 하는데 일관성이 떨어져서 가독성을 해칠 거 같아요 ..!

가 문제라면 아래처럼 해결 할 수 있지 않을까여?

isFormData 인자로 관리하는 대신, form data를 req할때는 인스턴스에 헤더를 추가하면 어떨까요?
```javascript
instance.post('/foo', {
    data:{ // ... },
    headers:{
       Contents-type:'multipart/form-data ...'
    }
})
```

Reply:
그렇네요 이 방법 괜찮은 거 같아요..!

일단 PR부분 머지하고 이후에 리팩토링 진행해도 될까요?
FormData 부분 실제로 구현해보고 hooks로 빼보기도 하고 customAPI 리팩토링 해볼게요!!

Reply:
넵넵 그럼요!!! 지윤님 작업 막히지 않도록 approve 했습니다 🥹 수고하셨어요!!